pdf-2/src/main/java/com/baeldung/exceltopdf/ExcelToPDFConverter.java

```diff
@@ -0,0 +1,167 @@
+        PdfPTable table = new PdfPTable(worksheet.getRow(0).getPhysicalNumberOfCells());
+        table.setWidthPercentage(100);
+        addTableHeader(worksheet, table);
+        addTableData(worksheet, table);
+        document.add(table);
+    }
+
+    private static void addTableHeader(XSSFSheet worksheet, PdfPTable table) throws DocumentException, IOException {
+        Row headerRow = worksheet.getRow(0);
+        for (int i = 0; i < headerRow.getPhysicalNumberOfCells(); i++) {
+            Cell cell = headerRow.getCell(i);
+            String headerText = cell.getCellType() == CellType.STRING ? cell.getStringCellValue() : String.valueOf(cell.getNumericCellValue());
+            PdfPCell headerCell = new PdfPCell(new Phrase(headerText, getCellStyle(cell)));
+            headerCell.setHorizontalAlignment(Element.ALIGN_CENTER);
+            table.addCell(headerCell);
+        }
+    }
+
+    private static void addTableData(XSSFSheet worksheet, PdfPTable table) throws DocumentException, IOException {
+        Iterator<Row> rowIterator = worksheet.iterator();
+        while (rowIterator.hasNext()) {
+            Row row = rowIterator.next();
+            if (row.getRowNum() == 0) {
+                continue;
+            }
+            for (int i = 0; i < row.getPhysicalNumberOfCells(); i++) {
+                Cell cell = row.getCell(i);
+                String cellValue;
+                if (cell != null) {
+                    if (cell.getCellType() == CellType.STRING) {
+                        cellValue = cell.getStringCellValue();
+                    } else if (cell.getCellType() == CellType.NUMERIC) {
+                        cellValue = String.valueOf(cell.getNumericCellValue());
+                    } else {
+                        cellValue = "";
+                    }
+                } else {
+                    cellValue = "";
+                }
+                PdfPCell cellPdf = new PdfPCell(new Phrase(cellValue, getCellStyle(cell)));
+                // Set background color
+                short bgColorIndex = cell.getCellStyle()
+                    .getFillForegroundColor();
+                if (bgColorIndex != IndexedColors.AUTOMATIC.getIndex()) {
+                    XSSFColor bgColor = (XSSFColor) cell.getCellStyle()
+                        .getFillForegroundColorColor();
+                    if (bgColor != null) {
+                        byte[] rgb = bgColor.getRGB();
+                        if (rgb != null && rgb.length == 3) {
+                            cellPdf.setBackgroundColor(new BaseColor(rgb[0] & 0xFF, rgb[1] & 0xFF, rgb[2] & 0xFF));
+                        }
+                    }
+                }
+
+                cellPdf.setHorizontalAlignment(Element.ALIGN_CENTER);
+                table.addCell(cellPdf);
+            }
+        }
+    }
+
+    private static Font getCellStyle(Cell cell) throws DocumentException, IOException {
```

Comment:
The article says that we can preserve the cell style including the font, borders, colors, etc. However, the code has the font and font size hardcoded. Can we either change the code or the description in the article? Otherwise, it's confusing to the reader.

Reply:
Noted on this, I'll update the code, and in the description will also highlight the limited font family iText support.